Examples/ExamplesApp/MenuItem.swift

```diff
@@ -0,0 +1,17 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+protocol MenuItem {
```

Comment:
This would make more sense as an enum with associated types since the concrete type is needed and the set of conforming types is known and finite. Also, a better name would be `ListItem` as the items are used in a list, not a menu.

That being said, I found this new design confusing as I was testing it. One contributing factor to that is that sidebars do not display disclosure indicators, so there is no visual distinction between an example and a category. I had no idea whether what I was selecting would open an example or push another list on the navigation stack.

Reply:
Would using a DisclosureGroup be better? Since we're switching to a (mostly) flat list, that may be more inline with what the rest of it.

Reply:
Yes, I think that would be better and more in line with what HIG says regarding sidebars.

Reply:
> This would make more sense as an enum with associated types...

I had looked at doing that initially but felt a protocol simplified things a good amount and allowed us to continue declaring each example like this:

https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/2807fe5579bdcc235c575027178636ed957a5454/Examples/ExamplesApp/Examples.swift#L90

instead of this or something similar:

https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/f3506b06f22ff21a480047c0129c8935d9d516e1/Examples/ExamplesApp/Examples.swift#L66

Regardless, I've changed to the enum and am okay with proceeding with it.